### DISCUSSION TITLE
cli_helper payload limit

### DISCUSSION THREAD

**Discussion by pfw** [koordinates/kart#770] [Q&A]
@olsen232 Did you get the whatever issue you ran across with the payload limit in the cli helper sorted? There was originally code around filter the environment for dealing with something along those lines - particularly running within a debugger environment could end up with a pretty massive environment which would truncate when passed through.

**Comment by olsen232** [comment]
Yeah I just hacked it to stitch payloads together until it finds one that ends with a `}`, which is probably robust enough for now.
There's some other issue though - when the long-running cli helper process tries to spawn a subprocess using `asyncio.create_subprocess_exec`, there seems to be some issue with the file-descriptors of that subprocess. At least, on linux.

**Reply by pfw** [reply]
There could be something where the calling process sets up FDs not on 0,1,2 and so the code at https://github.com/koordinates/kart/blob/410ee851a8e8ec826e7fe4ad5e7a31b0f1804dfa/cli_helper/kart.c#L130 messes with the wrong things. The actual FDs passed to the long running server are collected at https://github.com/koordinates/kart/blob/410ee851a8e8ec826e7fe4ad5e7a31b0f1804dfa/cli_helper/kart.c#L100 and that doesn't reference the int values 0,1,2 directly.

Possibly something around that - it was a little fiddly to get right in the first place.
